src/browserlib/map-ids-to-headings.mjs

```diff
@@ -114,8 +117,12 @@ export default function () {
       if (id) {
         mapping.id = id;
       }
+      alternateIds = alternateIds.filter(id => id !== mapping.id);
```

Comment:
Instead of filtering out, I would find it more natural to have `alternateIds` really be the list of alternate IDs from the start. Unless I missed something, that seems doable if you:
- drop the `push` lines 96 and 103
- add something like `if (id) { alternateIds.push(id); }` line 108 before overriding `id`
- drop this line 120

Alternatively, the variable could be called `ids`. You'd push all ids, you'd pop the last one for use as `mapping.id`, and set the other one as `mapping.alternateIds` if there's a rest. Up to you :)

Reply:
I'm not sure I understood if you meant the final data should always have all ids, or if this was only about code structure; I've updated the code on the basis of the latter for now

Reply:
I confirm I meant the latter.